elasticProcessor.go

```diff
@@ -341,16 +348,28 @@ func (ei *elasticProcessor) SaveMiniblocks(header data.HeaderHandler, body *bloc
 // SaveTransactions will prepare and save information about a transactions in elasticsearch server
 func (ei *elasticProcessor) SaveTransactions(
 	body *block.Body,
-	header data.HeaderHandler,
-	txPool map[string]data.TransactionHandler,
-	selfShardID uint32,
+	header nodeData.HeaderHandler,
+	pool *indexer.Pool,
 	mbsInDb map[string]bool,
 ) error {
 	if !ei.isIndexEnabled(txIndex) {
 		return nil
 	}
 
-	txs, alteredAccounts := ei.prepareTransactionsForDatabase(body, header, txPool, selfShardID)
+	allTxs := make(map[string]nodeData.TransactionHandler)
+
+	sliceMaps := []map[string]nodeData.TransactionHandler{
+		pool.Txs, pool.Scrs, pool.Receipts, pool.Invalid, pool.Rewards,
+	}
+
+	for _, txsMap := range sliceMaps {
```

Comment:
This could have been another function "mergeSliceOfMaps"

Reply:
done.